src/Hint/InterpreterT.hs

```diff
@@ -163,7 +165,7 @@ initialState = St {
                    hintSupportModule = error "No support module loaded!",
                    importQualHackMod = Nothing,
                    qualImports       = [],
-                   defaultExts       = [],
+                   defaultExts       = error "defaultExts missing!",
```

Comment:
This one is tricky. It does seem obviously better to use `[]` than bottom. So obvious, in fact, that there is no way the original author didn't think of it. So I tried to figure out why this `error` is here, and I think I figured it out: `[]` has the right type, but it's not a valid value for this field! It normally contains a bunch of information about tons of extensions, and I don't know what would happen if the code which looks for that information found an empty list instead. Of course, with or without your change, that code does _not_ find an empty list, because that field currently gets initialized to a correct value later on. But I think this `error` is there as a reminder to be careful to not accidentally leave that field uninitialized (or initialized to `[]`), because that could lead to trouble elsewhere. I think. I haven't been able to find what exactly makes use of that long list of information and what would happen if it didn't find it.

Of course, since that bottom is a _runtime_ error, it's not actually much of a reminder at all. Ideally this code should be refactored so we get a statically-checked reminder, not a runtime error, but at least a runtime error is better than silently returning the wrong answer.

Reply:
I've figured out what would go wrong if we forgot to initialize that field and accidentally left it as `[]`. The following block:

https://github.com/haskell-hint/hint/blob/b640d6ddc08547f416f3d3071cc37e55af5a15e1/src/Hint/Configuration.hs#L88-L89

is supposed to reset all the language extensions to their default state, but if the list is empty, it does nothing, leaving all the language extensions the way they are. Thus, in the following test, the last `do_load` would succeed when it should fail.

https://github.com/gelisam/hint/blob/b640d6ddc08547f416f3d3071cc37e55af5a15e1/unit-tests/run-unit-tests.hs#L55-L64